pydantic/root_model.py

```diff
@@ -132,11 +132,17 @@ def model_dump(  # type: ignore
             exclude_none: bool = False,
             round_trip: bool = False,
             warnings: bool = True,
-        ) -> RootModelRootType:
+        ) -> Any:
             """This method is included just to get a more accurate return type for type checkers.
             It is included in this `if TYPE_CHECKING:` block since no override is actually necessary.
 
             See the documentation of `BaseModel.model_dump` for more details about the arguments.
+
+            Generally, this method will have a return type of `RootModelRootType`, assuming that `RootModelRootType` is
+            not a `BaseModel` subclass. If `RootModelRootType` is a `BaseModel` subclass, then the return
+            type will likely be `dict[str, Any]`, as `model_dump` calls are recursive. The return type could
+            even be something different, in the case of a custom serializer on a `BaseModel` subclass.
```

Comment:
or a custom serializer on the root annotation. I think you can just say "in the case of a custom serializer".

Reply:
```suggestion
            even be something different, in the case of a custom serializer.
```